bind/gencffi_type.go

```diff
@@ -196,3 +281,32 @@ func (g *cffiGen) genTypeTPStr(sym *symbol) {
 	g.wrapper.Outdent()
 	g.wrapper.Printf("\n")
 }
+
+// genTypeSetItem generates __setitem__ of a type.
+func (g *cffiGen) genTypeSetItem(sym *symbol) {
+	g.wrapper.Printf("def __setitem__(self, idx, item):\n")
+	g.wrapper.Indent()
+	g.wrapper.Printf("if idx >= len(self):\n")
+	g.wrapper.Indent()
+	g.wrapper.Printf("raise IndexError('array index out of range')\n")
```

Comment:
ditto: it would be much nicer to raise an error with the correct type name (array/slice).